django/db/backends/utils.py

```diff
@@ -74,6 +76,15 @@ def executemany(self, sql, param_list):
         )
 
     def _execute_with_wrappers(self, sql, params, many, executor):
+        if not apps.ready:
+            warnings.warn(
+                "Accessing the database during app initialization is discouraged. "
```

Comment:
@adamchainz @felixxm I reverted the changes to `django.setup()` and now simply check whether the app registry is ready in the `CursorWrapper`. Should I be doing this in `callproc()` above as well?

Reply:
> @adamchainz @felixxm I reverted the changes to `django.setup()` and now simply check whether the app registry is ready in the `CursorWrapper`. Should I be doing this in `callproc()` above as well?

Please do :+1:

Reply:
I've added the warning to `callproc()` as well and moved the existing warning down into the `_execute()` and `_executemany()` methods so that there's the same structure in `callproc()`, `_execute()`, and `_executemany()` now:

https://github.com/django/django/blob/926f2b3511f92fbd5fab1b7f77ecf021d5ee2d68/django/db/backends/utils.py#L85-L88

However, this means that the warning wouldn't be triggered if there's an execute wrapper that prevents the query from actually executing, which feels okay to me. Or would you prefer to always trigger the warning regardless of what the execute wrappers are doing?

Reply:
The issue to prevent is actual database queries, which slow down import time and can be cached forever. If an execute wrapper prevents the query, those concerns are avoided, so yes, it also seems fine to me to ignore that case.